trace/Sampling.md

```diff
@@ -8,7 +8,31 @@ exporters.
 
 The Sampling bit is always set only at the start of a Span, using a `Sampler`
 
-### What kind of samplers does OpenCensus support?
+### Sampler interface
+A `Sampler` is an interface that returns a `SamplingDecision` based on given sampling parameters.
+
+#### Sampling parameters
+Based on the language the sampling parameters can be embedded into an object called
+`SamplingParameters` or explicitly passed as arguments. The set of sampling parameters SHOULD
+include at least:
```

Comment:
what's the mechanism for sampler to set a sampling flag for downstream components? Or sampling score be added in `tracestate`? Or who's responsibility is it? Should propagator deal with this? Worth mentioning here

Reply:
I feel it is a race condition here, what is the order of these events:
* Modify the Tracestate
* Call sampler.

If we call sampler first, we cannot use the current Tracestate (only parent Tracestate) in the decision. If we create the Tracestate first we cannot modify it after the Sampler is called (or I guess we can create it again).

What is your thought on the order of these two events?

Reply:
@lmolkova any update on this?

Reply:
I don't have any pictures about it. 

My take on this:

tracing system should have a chance to modify tracestate before sampler makes a decision to support scenarios where sampling is controlled with tracestate (mentioned by Sergey).

I don't see an obvious scenario when tracestate is modified after sampling decision is made or span is started. Do we really want to support this?
Tracing system is the only thing that should touch tracestate and it should have just one opportunity to do so: after span is created, so I don't see how we can have a race condition here.

Reply:
So I think what I called "race condition" is not the best term. Here are the sequence of events that I can see in time order:
1. Tracestate is modified before the sampler gets called as you suggested.
2. Sampler gets called. 
3. Tracestate is modified with sampling results - I heard from @SergeyKanzhelev that here some systems may want to modify the trace state with results from the sampling

Do I miss something?

Reply:
I also often accidentally say "race condition" when mean "chicken egg"

Though I wonder if tracestate needs to be exposed in-process or not. For example, if it is only expected to be written on remote links, is it necessary for us to write our own here? For example, it seems necessary to write upstream, but possibly not to write our own trace state until the edge where it is exported to headers..

Reply:
@lmolkova @SergeyKanzhelev any conclusion on this? I would like to finish this document.